kick_fifo.asm

```diff
@@ -2,49 +2,74 @@
 
 bits 32
 
+%define CACHE_PUSH_MASTER_STATE     0xFD003200
+%define CACHE_PUSH_STATE            0xFD003220
+%define DMA_PUSH_ADDR               0xFD003240
+%define DMA_PULL_ADDR               0xFD003244
+
 kick_fifo:
 
-; Load expected_put into reg
-mov edx, dword [esp+4]
+push ebp
+mov ebp, esp
+push ebx
```

Comment:
These prologues / epilogues should not be necessary because xboxpy `CALL` will store / restore all general-purpose registers.